ingress-watcher/cmd/push.go

```diff
@@ -0,0 +1,90 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/cybozu-go/log"
+	"github.com/cybozu-go/well"
+	"github.com/cybozu/neco-containers/ingress-watcher/pkg/watch"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var pushConfig struct {
+	JobName      string
+	PushAddr     string
+	PushInterval time.Duration
+}
+
+var pushCmd = &cobra.Command{
+	Use:   "push",
+	Short: "Push metrics to Pushgateway",
+	Long:  `Push metrics to Pushgateway`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if configFile != "" {
+			if err := viper.Unmarshal(&pushConfig); err != nil {
+				return err
+			}
+		}
+
+		if len(pushConfig.JobName) == 0 {
```

Comment:
Couldn't you use MarkFlagRequired?

Reply:
This is because I cannot find the way to set flag as required when the value can be given from both YAML file and command line flag.

For example, when I set MarkFlagRequired and configure a value with YAML file, viper gets an error which says the value is not set even if the value is definitely set.